tests/trainer/test_optimizers.py

```diff
@@ -311,25 +348,125 @@ def test_configure_optimizers_with_frequency(tmpdir):
     model = EvalModelTemplate()
     model.configure_optimizers = model.configure_optimizers__multiple_optimizers_frequency
 
-    trainer = Trainer(
-        default_root_dir=tmpdir,
-        max_epochs=1
-    )
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
+    result = trainer.fit(model)
+    assert result
+
+
+def test_configure_optimizers_with_frequency_and_lr_schedulers(tmpdir):
+    """
+    Test that opt_idx is set for LR schedulers when corresponding frequency is set for multiple optimizers.
+    """
+
+    class DummyModel(BoringModel):
+        def configure_optimizers(self):
+            optimizer1 = optim.Adam(self.parameters(), lr=0.01)
+            optimizer2 = optim.Adam(self.parameters(), lr=0.01)
+
+            lr_dict = {
+                'scheduler': optim.lr_scheduler.OneCycleLR(optimizer2, max_lr=0.01, total_steps=1),
+                'interval': 'step',
+            }
+
+            return [
+                {'optimizer': optimizer1, 'frequency': 5},
+                {
+                    'optimizer': optimizer2,
+                    'frequency': 1,
+                    'lr_scheduler': lr_dict,
+                },
+            ]
+
+    model = DummyModel()
+    trainer = Trainer(default_root_dir=tmpdir, max_epochs=1)
+    _, lr_schedulers, _ = trainer.init_optimizers(model)
+    assert lr_schedulers[0]['opt_idx'] == 1
+
+
+def test_step_scheduling_for_multiple_optimizers_with_frequency(tmpdir):
```

Comment:
can you add some strong assertions for what the actual learning rate value should be in the end after training with frequency < limit_train_batches.

Reply:
Stronger assertions would create a dependency on the PyTorch implementation of the  LR schedulers since you'll have to match the LR rate exactly with the implementation. This dependency can cause failures down the line if either the defaults or the implementation changes in Pytorch. 

The assertion here checks that the LR scheduler works as expected because if it's called any more than the number of steps specified, it'll throw an error.

Reply:
ok. noticed one more thing. with `limit_train_batches=5` it doesn't call that scheduler. I'd suggest frequency here to be 2, 3 repectively with `limit_train_batches=5`, `total_steps=3` and add an assertion with `assert trainer.lr_schedulers[0]['scheduler']._step_count == 4`. This will make the test strong.

Reply:
Added the stronger assertions. Changed the tests so that one checks two schedulers and other checks just one. 
Also discovered an interesting bug - `if lr_scheduler['opt_idx']` returns `False` if that value is 0 so that specific scheduler will always get updated.

Reply:
this is weird. why is that? did you find a reason?

Reply:
I guess `0` resolves to `False`. `int(0).__bool__()` is `False`